package/MDAnalysis/lib/util.py

```diff
@@ -534,39 +535,6 @@ def isstream(obj):
     return np.any(alternative_results)
 
 
-def which(program):
```

Comment:
So the idea here is not to remove `which` but deprecate it, since there might be downstream stuff that still uses it.

Reply:
Also once you re-instate this code & add the deprecation decorator, can you add a tests that checks the deprecation warnings please?

Reply:
> So the idea here is not to remove `which` but deprecate it, since there might be downstream stuff that still uses it.

As per @orbeckst's comment [url](https://github.com/MDAnalysis/mdanalysis/issues/3649#issuecomment-1126864675), the function body of which was supposed to contain a function call to shutil.which.

Reply:
No by that he means that the calls to `utils.which` in other parts of the code need to be switched to `shutil.which`, not the existing `utils.which` method

Reply:
@IAlibay explained below.

Reply:
@IAlibay explained below.